pkg/web/http.go

```diff
@@ -0,0 +1,69 @@
+package web
+
+import (
+	"bytes"
+	"crypto/tls"
+	"github.com/gorilla/websocket"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"time"
+)
+
+type WrappedSocketListener struct {
```

Comment:
WrappedSocketListener cannot be reused after being closed, correct?  If so, add a `closed int32` field to this struct to signal if Close has already been called, and if close the done channel.